common/scala/transformEnvironment.sh

```diff
@@ -34,6 +34,12 @@ configVariables=$(compgen -v | grep $prefix)
 
 props=()
 
+if [ -n "$OPENWHISK_CONFIG" ]
+then
+   printf "%s" "$OPENWHISK_CONFIG" > "/openwhisk.conf"
+   props+=("-Dconfig.file='/openwhisk.conf'")
+fi
```

Comment:
Can we generalize this a bit? As of now, this script is very non-OpenWhisk-dependent. What about:

```bash
if [ -n "$CONFIGFILE_CONTENT" ]
then
    location="/config.conf"
    echo "$CONFIGFILE_CONTENT" > "$location"
    props+=("-Dconfig.file='$location'")
fi
```

Note: I switched from `printf` to `echo`. Any reason to use `printf` here?
Note 2: Please fix the indenting from 3 to 4 spaces.

Reply:
Used `printf` based on [this answer][1] on SO.

[1]: https://stackoverflow.com/a/49418406/1035417

Reply:
> CONFIGFILE_CONTENT

I preferred `OPENWHISK_CONFIG` as it makes it explicit that its an OpenWhisk related config and not generic one. So once we document its usage it would make it obvious to end user.

Reply:
As I said, personally I didn't like making this file OpenWhisk specific. I guess that's a very minor thing anyway though.

`printf` seems sensible thanks.